spec/requests/reports/distributions_summary_requests_spec.rb

```diff
@@ -0,0 +1,95 @@
+require "rails_helper"
+
+RSpec.describe "Distributions", type: :request do
+  context "while signed in" do
+
+    before do
+      travel_to('2024-01-17')
+    end
+
+    before do
+      sign_in(@user)
+    end
+
+    context "the index page" do
+
+      context "without filters" do
+        before do
+          get reports_distributions_summary_path(@user.organization)
+        end
+
+        it "shows a list of recent distributions" do
+          expect(response.body).to include("Recent distributions")
+        end
+
+        it "has a link to create a new distribution" do
+          expect(response.body).to include("distributions/new")
+        end
+      end
+
+      context "with filters" do
+        before do
+          # Create a bunch of historical distributions
+          create :distribution, :with_items, item_quantity: 2, issued_at: 0.days.ago
+          create :distribution, :with_items, item_quantity: 3, issued_at: 1.days.ago
+          create :distribution, :with_items, item_quantity: 7, issued_at: 3.days.ago
+          create :distribution, :with_items, item_quantity: 11, issued_at: 10.days.ago
+          create :distribution, :with_items, item_quantity: 13, issued_at: 20.days.ago
+          create :distribution, :with_items, item_quantity: 17, issued_at: 30.days.ago
+        end
+
+        let(:formatted_date_range) { date_range.map { _1.to_formatted_s(:date_picker) }.join(" - ") }
+
+        before do
+          get reports_distributions_summary_path(@user.organization), params: { filters: { date_range: formatted_date_range } }
+        end
+
+        context "today" do
```

Comment:
@dorner what do you think about this as a replacement for the similar system specs? They are much more explicit and get at the point, which shouldn't be to test every possible filter but rather to test that filtering works in general. The filtered values come through as date ranges in any case.

I am thinking I'll replicate this pattern for the other handful of reports. It is still repetitive, though I kinda like repetitive specs.

Reply:
Welllll I went ahead and made them all like this, trying to capture the most important assertions of the system specs.

Reply:
I like it! Definitely not necessary to do this over and over again for everything that has filters.